lib/client-assets.php

```diff
@@ -866,9 +866,8 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Prepare Jed locale data.
 	$locale_data = gutenberg_get_jed_locale_data( 'gutenberg' );
 	wp_add_inline_script(
-		'wp-edit-post',
-		'wp.i18n.setLocaleData( ' . json_encode( $locale_data ) . ' );',
-		'before'
+		'wp-i18n',
+		'wp.i18n.setLocaleData( ' . json_encode( $locale_data ) . ' );'
```

Comment:
Should we move this inline script to line 140 (registration time)?

Reply:
No, because `gutenberg_editor_scripts_and_styles()` gets only called if Gutenberg gets initialized while `gutenberg_register_scripts_and_styles()` is called on all pages. Also, the strings are not used by `wp-i18n`.

Reply:
That's exactly the point I was raising before, what if a plugin adds a dependency to `wp-components` or `wp-blocks` in a separate page? The strings should be loaded without any other adjustment from the plugin author.

Reply:
It's a fundamental problem with the current API. Based on https://github.com/WordPress/packages/pull/96#issuecomment-378529253, "there's no blocking here. So moving forward.". 🤷‍♂️

Reply:
@aduth Thoughts on this

Reply:
It will need further changes for merge. I imagine it might be the domain-per-script idea you already mentioned, ideally with some automation for assigning domain to a script bundle for string extraction and injecting the locale data inline script. Admittedly unclear how well this fits into existing localization workflows.

As for what to do here, I don't feel strongly _against_ this implementation as a short-term solution, but I agree the locale data should ideally be set as a side effect of depending on any of the internationalized Gutenberg scripts. I see this as a stubbed script upon which all of the Gutenberg bundles depend, with the inline script to set locale data. This was already mentioned, though dismissed as complicated. Personally it doesn't seem very complicated to me 🤷‍♂️